src/shims/backtrace.rs

```diff
@@ -154,4 +230,31 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
 
         Ok(())
     }
+
+    fn handle_miri_resolve_frame_names(
+        &mut self,
+        abi: Abi,
+        link_name: Symbol,
+        args: &[OpTy<'tcx, Tag>],
+    ) -> InterpResult<'tcx> {
+        let this = self.eval_context_mut();
+
+        let &[ref ptr, ref flags, ref name_ptr, ref filename_ptr] =
+            this.check_shim(abi, Abi::Rust, link_name, args)?;
+
+        let flags = this.read_scalar(flags)?.to_u64()?;
+        if flags != 0 {
+            throw_unsup_format!("unknown `miri_backtrace_size` flags {}", flags);
+        }
+
+        let (fn_instance, lo) = this.resolve_frame_pointer(ptr)?;
+
+        let name = fn_instance.to_string();
+        let filename = lo.file.name.prefer_remapped().to_string();
```

Comment:
This is duplicated across the two users of `resolve_frame_pointer`, can you move it into `resolve_frame_pointer`?